_includes/story.md

```diff
@@ -227,61 +233,69 @@ How can we build a BJP that will maximize the use of available hardware resource
+  - These jobs would take `6.91M * 26B = 171.34 MB` of memory in order to be stored in Redis. At this rate, our jobs storage space would be reaching the storage limit after 24 (4096 / 171.34) days.
 
-An additional backlog of 96 hours of email sending jobs everyday, along with an additional 1.17GB of memory occupied by this same backlog of email sending jobs every week make for a very inefficient background job system.
+With our current naive approach we have the **latency increasing at a rate of 96 hours per day**, along with the **jobs storage space increasing at a rate of 171.34 MB per day**. This would make for a very inefficient background job system.
 
-The challenge for us here is to even the enqueuing throughput and the dequeuing throughput. We should not slow down the enqueuing side or else our web application will become unresponsive, waiting for jobs to be enqueued before moving on, so let's focus on increasing the processing throughput.
+The challenge for us here is to balance out the enqueuing throughput and the dequeuing throughput. Slowing down the enqueuing side is not a viable option because this web application would become unresponsive, waiting for jobs to be enqueued before moving on, so let's focus on increasing the processing throughput.
 
 #### Concurrency & Threads
 
-Based on our scenario, we need to maximize the job processing through put by a factor of 5. How can we increase the job processing throughput when it comes to a BJP?
+Based on our scenario, we need to maximize the job processing throughput by a factor of 5. How can this be achieved?
+
+Currently, we have one worker processing the jobs one by one, sequentially. Our goal is to process these jobs 5 times as fast, hence, we need five workers instead of one, working together so they can process five jobs, instead of one, in 50 ms. This is where the concept of **Concurrency** comes in.
 
-Currently, we have one worker processing the jobs one by one, sequentially. Our goal is to process 5 these jobs 5 times as fast, this means we could have 5 workers instead of 1 working together in order to process 5 jobs, instead of one, in 50 ms. This is where the concept of **Concurrency** comes in.
-Concurrency is employed to design atomic units of work than can be executed at the same time, concurrently. This means that for a time window, two or more units of works are worked on, it could be at the exact same time (in parallel) or alternatively but never at the exact same time (not in parallel). In our case, units of works would be the jobs that need to be processed.
+Concurrency is employed to design atomic units of work than can be executed at the same time, concurrently. That is, for a given time window, two or more units of work are worked on. It could be at the exact same time (in parallel), or, alternatively but never at the exact same time (not in parallel). In our case, a unit of work is a job.
```

Comment:
I think the definition as is good. What you suggest is actually wrong: `but never at the exact same time`. They can be executed in parallel and concurrently, one does not prevent the other.

Reply:
But here we are talking about concurrency, not parallelism. And that means, by definition, that code cannot execute at the exact same time.

Reply:
Yes it can, concurrency means that the code is designed in atomic units of work that can be executed separately, either in parallel or sequentially. This last part doesn't matter to concurrency, it's more about designing these atomic units of work so that they can be executed separately.

Reply:
OK, then I misunderstood that point. But the text still needs some refactoring. How about this:
`By using concurrent execution of code we effectively introduce atomic units of work that can be run at the same time. That is, for a given time window, two or more units of work can be processed at the same time (in parallel) or, alternatively, at the overlapping time periods (not in parallel). In our case, a unit of work is a job.`

Reply:
I feel like that takes away from explaining concurrency and just immediately jumping into why concurrency is useful. Here is a refactored version, actually simplifying things:

> Concurrency is employed to design atomic units of work that can be executed independently from one another, effectively enabling them to potentially be executed at the same time (in parallel or not). In our case, a unit of work is a job.